### DISCUSSION TITLE
Get slug field of other languages of the same document in the same request

### DISCUSSION THREAD

**Discussion by wiesson** [payloadcms/payload#2386] [Q&A]
My page has three languages and I'd like to achieve two tasks: 

1. Add the `hreflang` url of the other languages for SEO
2. If the user switches the language, use the correct slug to switch to the "same page" in the desired language (if that page is available)

The problem is that each page has a different slug so I have to query the first page by slug, then use the id to query the others.

```HTML
<link rel="canonical" href="https://www.mypage.com/en/europe-trips">
<link rel="alternate" href="https://www.mypage.com/europa-reisen" hreflang="de">
<link rel="alternate" href="https://www.mypage.com/en/europe-trips" hreflang="en">
<link rel="alternate" href="https://www.mypage.com/fr/voyager-en-europe" hreflang="fr">
```

In the above mentioned example, I need two requests to obtain DE/FR (from EN perspective). The same applies to subpages, of that page, because I need the other slug of the entry itself and their parent.

Question: Is there a simpler method so that I don't have so many queries (at least 2-3 per page and language)?

Best solution to me would if each document would return their "siblings" automatically. Could this be achieved by a hook?

**Comment by qbeauperin** [comment]
You can pass `locale=*` or `locale=all` in your query to get all languages. 👍 

From [the docs](https://payloadcms.com/docs/configuration/localization#retrieving-localized-docs:~:text=The%20REST%20and%20Local%20APIs%20can%20return%20all%20localization%20data%20in%20one%20request%20by%20passing%20%27all%27%20or%20%27*%27%20as%20the%20locale%20parameter.):
> The REST and Local APIs can return all localization data in one request by passing 'all' or '*' as the locale parameter.

**Reply by wiesson** [reply]
How did I not know this 🤯

Thank you!

**Comment by r1tsuu** [comment]
A bit late, but i did this:

```ts
export const slugBeforeRead: BeforeReadHook = async ({ doc, req }) => {
  return {
    ...doc,
    slugs: doc.slug,
  };
};
```
```ts
export const BlogCategories: CollectionConfig = {
  slug: "blog-categories",
  access: {
    read: () => true,
  },
  labels: {
    plural: "Blog categories",
  },
  hooks: {
    beforeRead: [slugBeforeRead],
  },
```

So now in API you also have "slugs" field contain all languages with 1 request only, if you use locale=* you still have to fetch for your language

**Reply by wiesson** [reply]
Wonderful! It looks cleaner than my current solution! I'll try that later!
